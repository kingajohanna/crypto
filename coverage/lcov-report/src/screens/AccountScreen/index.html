
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/screens/AccountScreen</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/screens/AccountScreen</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.22% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>34/72</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">10.52% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>2/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.76% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>32/67</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="UserLoggedIn.tsx"><a href="UserLoggedIn.tsx.html">UserLoggedIn.tsx</a></td>
	<td data-value="80" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 80%"></div><div class="cover-empty" style="width: 20%"></div></div>
	</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="10" class="abs high">8/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="9" class="abs medium">7/9</td>
	</tr>

<tr>
	<td class="file low" data-value="UserNotLoggedIn.tsx"><a href="UserNotLoggedIn.tsx.html">UserNotLoggedIn.tsx</a></td>
	<td data-value="41.93" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.93" class="pct low">41.93%</td>
	<td data-value="62" class="abs low">26/62</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="22" class="abs low">0/22</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="16" class="abs low">1/16</td>
	<td data-value="43.1" class="pct low">43.1%</td>
	<td data-value="58" class="abs low">25/58</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu Apr 28 2022 12:48:39 GMT+0200 (GMT+02:00)
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    