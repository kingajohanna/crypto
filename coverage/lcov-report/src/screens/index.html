
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">72.28% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>60/83</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">55% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>11/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">40.9% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>9/22</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">68.91% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>51/74</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="AccountScreen.tsx"><a href="AccountScreen.tsx.html">AccountScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file high" data-value="FavouritesScreen.tsx"><a href="FavouritesScreen.tsx.html">FavouritesScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="MarketScreen.tsx"><a href="MarketScreen.tsx.html">MarketScreen.tsx</a></td>
	<td data-value="54" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 54%"></div><div class="cover-empty" style="width: 46%"></div></div>
	</td>
	<td data-value="54" class="pct medium">54%</td>
	<td data-value="50" class="abs medium">27/50</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	<td data-value="51.06" class="pct medium">51.06%</td>
	<td data-value="47" class="abs medium">24/47</td>
	</tr>

<tr>
	<td class="file high" data-value="MyCoinsScreen.tsx"><a href="MyCoinsScreen.tsx.html">MyCoinsScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu Apr 28 2022 12:48:39 GMT+0200 (GMT+02:00)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    