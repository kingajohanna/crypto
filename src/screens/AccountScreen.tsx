import React from "react";

import { Tabs } from "@constants/Tabs";

import { AccountButton } from "@components/AccountButton";
import { ScreenBackground } from "@components/ScreenBackground";

import auth from "@react-native-firebase/auth";
import { GoogleSignin } from "@react-native-google-signin/google-signin";

GoogleSignin.configure({
    webClientId: "846343653708-2vjl1i736qqn5lgfqia80g2ieqsm3ktj.apps.googleusercontent.com",
});

async function onGoogleButtonPress() {
    // Get the users ID token
    const { idToken } = await GoogleSignin.signIn();

    // Create a Google credential with the token
    const googleCredential = auth.GoogleAuthProvider.credential(idToken);

    // Sign-in the user with the credential
    const user = auth().signInWithCredential(googleCredential);

    user.then((re) => {
        console.log(re);
    });

    return user;
}

async function OnEmail() {
    return auth()
        .signInAnonymously()
        .then(() => {
            console.log("User signed in anonymously");
        })
        .catch((error) => {
            if (error.code === "auth/operation-not-allowed") {
                console.log("Enable anonymous in your firebase console.");
            }

            console.error(error);
        });
}

export const AccountScreen = () => {
    return (
        <ScreenBackground title={Tabs.account}>
            <AccountButton.Email onPress={() => OnEmail()} />
            <AccountButton.Apple />
            <AccountButton.Facebook />
            <AccountButton.Google onPress={() => onGoogleButtonPress().then(() => console.log("Signed in with Google!"))} />
            <AccountButton.Logout />
            <AccountButton.Delete />
        </ScreenBackground>
    );
};
