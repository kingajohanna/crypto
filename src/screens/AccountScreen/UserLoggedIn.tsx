import React, { useState } from "react";
import { AccountButton } from "@components/AccountButton";
import { DeleteAlert, LogoutAlert } from "@components/Alert";

/*
    account logout and delete buttons
*/
export const UserLoggedIn = () => {
    const [deletePopup, setDeletePopup] = useState(false);

    return (
        <>
            <AccountButton.Logout onPress={() => LogoutAlert()} />
            <AccountButton.Delete onPress={() => setDeletePopup(true)} />
            <DeleteAlert visible={deletePopup} setVisible={setDeletePopup} />
        </>
    );
};
