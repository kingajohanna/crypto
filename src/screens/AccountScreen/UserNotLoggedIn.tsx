import React, { useRef, useState } from "react";
import RBSheet from "react-native-raw-bottom-sheet";
import { shallowEqual, useSelector } from "react-redux";
import { AccountBottomModal } from "@components/AccountBottomModal";
import { AccountButton } from "@components/AccountButton";
import { TextInput } from "@components/InputField";
import { googleSignIn } from "@services/FirebaseServices";
import { Colors, hexToRGBA } from "@theme/Colors";
import { getModalHeight } from "@utils/BottomModalHeight";
import { setAuthErrorAction } from "@actions/ErrorActions";
import { RootState } from "@stores/store";
import { Text } from "@components/Text";
import { onLoginValidationAndLogin, onPasswordResetValidationAndReset, onSignUpValidationAndSignup } from "@utils/validation";

/*
    login and register screen
*/
export const UserNotLoggedIn = () => {
    const signUp = useRef() as React.MutableRefObject<RBSheet>;
    const logIn = useRef() as React.MutableRefObject<RBSheet>;
    const error = useSelector((state: RootState) => state.error.authError, shallowEqual);

    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [passwordConf, setPasswordConf] = useState("");

    const resetStates = () => {
        setEmail("");
        setPassword("");
        setPasswordConf("");
        setAuthErrorAction("");
    };

    const closeModal = () => {
        logIn.current?.close();
        signUp.current?.close();
        resetStates();
    };

    const onSignUp = async () => {
        const success = await onSignUpValidationAndSignup(email, password, passwordConf);
        if (success) closeModal();
    };

    const onLogin = async () => {
        const success = await onLoginValidationAndLogin(email, password);
        if (success) closeModal();
    };

    const onPasswordReset = async () => {
        await onPasswordResetValidationAndReset(email);
    };

    const onLoginPress = () => {
        logIn.current!.open();
    };

    const onSignUpPress = () => {
        signUp.current!.open();
    };

    return (
        <>
            <AccountButton.Google onPress={() => googleSignIn()} />
            <AccountButton.EmailLogIn onPress={() => onLoginPress()} />
            <AccountButton.EmailSignIn onPress={() => onSignUpPress()} />
            <RBSheet ref={signUp} closeOnPressMask={true} onClose={() => resetStates()} closeDuration={180} openDuration={180} height={getModalHeight(0.47)} customStyles={modalStyle}>
                <AccountBottomModal errorText={error} primaryButtonText="Register" primaryButtonOnPress={() => onSignUp()} secondaryButtonText="Cancel" secondaryButtonOnPress={() => closeModal()}>
                    <TextInput placeholder="Email address" title="Email" onChangeText={setEmail} value={email} />
                    <TextInput placeholder="Password" title="Password" onChangeText={setPassword} value={password} isSecret />
                    <TextInput placeholder="Confirm password" title="Password" onChangeText={setPasswordConf} value={passwordConf} isSecret />
                </AccountBottomModal>
            </RBSheet>
            <RBSheet ref={logIn} closeOnPressMask={true} onClose={() => resetStates()} closeDuration={180} openDuration={180} height={getModalHeight(0.4)} customStyles={modalStyle}>
                <AccountBottomModal errorText={error} primaryButtonText="Login" primaryButtonOnPress={() => onLogin()} secondaryButtonText="Cancel" secondaryButtonOnPress={() => closeModal()}>
                    <TextInput placeholder="Email address" title="Email" onChangeText={setEmail} value={email} />
                    <TextInput placeholder="Password" title="Password" onChangeText={setPassword} value={password} isSecret />
                    <Text onPress={() => onPasswordReset()}>Forgot password</Text>
                </AccountBottomModal>
            </RBSheet>
        </>
    );
};

const modalStyle = {
    container: {
        borderTopLeftRadius: 15,
        borderTopRightRadius: 15,
        backgroundColor: Colors.gunmetal,
    },
    wrapper: {
        backgroundColor: hexToRGBA(Colors.richBlack, 0.5),
    },
    draggableIcon: {
        backgroundColor: Colors.cadetBlue,
    },
};
