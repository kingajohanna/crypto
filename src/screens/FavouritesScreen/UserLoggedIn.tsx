import React, { useState } from "react";
import { ActivityIndicator, FlatList } from "react-native";
import { shallowEqual, useSelector } from "react-redux";
import RBSheet from "react-native-raw-bottom-sheet";
import { RootState } from "@stores/store";
import { MarketData } from "@constants/DataTypes";
import { getFavsMarket, getMarketData } from "@services/CryptoServices";
import { CryptoCoin } from "@components/CryptoCoin";
import { SelectedCoinGraph } from "@components/SelectedCoinGraph";

/**
 *    renders users favourite coins in favourites screen when the user is logged in
 */
export const UserLoggedIn = () => {
    const crypto = useSelector((state: RootState) => state.crypto, shallowEqual);
    const user = useSelector((state: RootState) => state.user, shallowEqual);

    const [loading, setLoading] = useState(false);

    const [selectedCoinData, setSelectedCoinData] = useState<MarketData | null>(null);

    const refRBSheet = React.useRef() as React.MutableRefObject<RBSheet>;

    const fetchFavData = async () => {
        await getFavsMarket(user.id);
        await getMarketData(user.id);
        setLoading(false);
    };

    const openModal = (item: MarketData) => {
        setSelectedCoinData(item);
        refRBSheet.current!.open();
    };

    const onRefresh = async () => {
        setLoading(true);
        await fetchFavData();
    };

    return (
        <>
            {loading && <ActivityIndicator animating size="large" style={{ paddingTop: 16 }} />}
            <FlatList
                keyExtractor={(item) => item.id}
                data={user.isLoggedIn ? crypto.favs : []}
                initialNumToRender={100}
                onRefresh={() => onRefresh()}
                refreshing={loading}
                getItemLayout={(data, index) => ({ length: 72, offset: 72 * index, index })}
                renderItem={({ item }) => (
                    <CryptoCoin
                        id={item.id}
                        isFavourite={item.fav}
                        onPress={() => openModal(item)}
                        name={item.name}
                        shortName={item.symbol}
                        price={item.current_price}
                        priceChange={item.price_change_percentage_7d_in_currency}
                        imageUrl={item.image}
                    />
                )}
            />
            <SelectedCoinGraph selectedCoinData={selectedCoinData!} reference={refRBSheet} />
        </>
    );
};
