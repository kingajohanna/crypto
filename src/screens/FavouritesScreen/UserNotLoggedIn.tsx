import React from "react";

import { UserRequiredContent } from "@components/UserRequiredContent";

export const UserNotLoggedIn = () => {
    return <UserRequiredContent />;
};
