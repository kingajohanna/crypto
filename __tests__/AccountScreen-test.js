import React from "react";
import * as renderer from "react-test-renderer";
import "jest-styled-components";
import ShallowRenderer from "react-test-renderer/shallow";
import { Provider } from "react-redux";
import { AccountScreen } from "../src/screens/AccountScreen";
import { UserLoggedIn } from "../src/screens/AccountScreen/UserLoggedIn";
import { UserNotLoggedIn } from "../src/screens/AccountScreen/UserNotLoggedIn";
import { store } from "../src/stores/store";

jest.useFakeTimers();

describe("Account Screen", () => {
    test("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <AccountScreen />
                </Provider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("UserLoggedIn renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <UserLoggedIn />
                </Provider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("UserNotLoggedIn renders correctly", () => {
        const renderer = new ShallowRenderer();
        const tree = renderer.render(
            <Provider store={store}>
                <UserNotLoggedIn />
            </Provider>,
        );

        expect(tree).toMatchSnapshot();
    });
});
