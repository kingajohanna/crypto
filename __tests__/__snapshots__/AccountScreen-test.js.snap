// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Screen UserLoggedIn renders correctly 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1F2833",
          "borderColor": "#45A29E",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 54,
          "marginTop": 12,
          "paddingHorizontal": 15,
          "width": 342,
        },
      ]
    }
    variant="logout"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginRight": 15,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#4CECE0",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4CECE0",
            "fontSize": 18,
            "lineHeight": 24,
          },
        ]
      }
      variant="logout"
    >
      Tap to logout
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#1F2833",
          "borderColor": "#45A29E",
          "borderRadius": 10,
          "borderWidth": 2,
          "flexDirection": "row",
          "height": 54,
          "marginTop": 12,
          "paddingHorizontal": 15,
          "width": 342,
        },
      ]
    }
    variant="delete"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "marginRight": 15,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#4CECE0",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4CECE0",
            "fontSize": 18,
            "lineHeight": 24,
          },
        ]
      }
      variant="delete"
    >
      Tap to delete account
    </Text>
  </View>,
  <View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onHide={[Function]}
      renderToHardwareTextureAndroid={true}
      transparent={true}
      useNativeDriver={false}
      visible={false}
    />
  </View>,
]
`;

exports[`Account Screen UserNotLoggedIn renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <UserNotLoggedIn />
</Context.Provider>
`;

exports[`Account Screen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1F2833",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#0B0C10",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1F2833",
            "borderBottomLeftRadius": 15,
            "borderBottomRightRadius": 15,
            "borderColor": "#1F2833",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C5C6C7",
              "fontSize": 24,
            },
          ]
        }
      >
        Account
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#45A29E",
            "borderRadius": 10,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 54,
            "marginTop": 12,
            "paddingHorizontal": 15,
            "width": 342,
          },
        ]
      }
      variant="google"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginRight": 15,
            },
          ]
        }
      >
        <Image
          source="test-file-stub"
          style={
            Object {
              "height": 24,
              "width": 24,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#8B8B8B",
              "fontSize": 18,
              "lineHeight": 24,
            },
          ]
        }
        variant="google"
      >
        Continue with Google
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1F2833",
            "borderColor": "#45A29E",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 54,
            "marginTop": 12,
            "paddingHorizontal": 15,
            "width": 342,
          },
        ]
      }
      variant="emailLogIn"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginRight": 15,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#4CECE0",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#4CECE0",
              "fontSize": 18,
              "lineHeight": 24,
            },
          ]
        }
        variant="emailLogIn"
      >
        Login with email
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#1F2833",
            "borderColor": "#45A29E",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 54,
            "marginTop": 12,
            "paddingHorizontal": 15,
            "width": 342,
          },
        ]
      }
      variant="emailSignIn"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginRight": 15,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#4CECE0",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#4CECE0",
              "fontSize": 18,
              "lineHeight": 24,
            },
          ]
        }
        variant="emailSignIn"
      >
        Register with email
      </Text>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    />
  </View>
</RCTSafeAreaView>
`;
